bzt/modules/shellexec.py

```diff
@@ -113,26 +118,50 @@ def check(self):
 
     def shutdown(self):
         for task in self.shutdown_tasks:
+            self._set_stop_reason_vars(task)
             task.start()
 
         for task in self.check_tasks + self.startup_tasks:
+            self._set_stop_reason_vars(task)
```

Comment:
does variables setup work for tasks started before? (the same question to post_process)

Reply:
I found no way to change env for tasks started in the past. And I see no need in that also.